### PR TITLE
Updates to build for Alpine

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

setdate.cpp was incorrectly assigning an int to a struct, fixed to initialize struct zero'd
update the typegen project in the bash script
added some help text at top of script for someone trying to build it
updated runtime to linux-musl-x64 instead of linux-x64 as Alpine distro uses the musl libc libraries instead of glibc which is used by most distros.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/libpsl-native/src/setdate.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ int32_t SetDate(struct private_tm* time)
 
 static int32_t GetTimeVal(struct private_tm& time, struct timeval& tv)
 {
-    struct tm nativeTime = 0;
+    struct tm nativeTime = {0};
```
[the remark this answers is not in the document]
Comment (SteveL-MSFT) [reply src/libpsl-native/src/setdate.cpp]: If we eventually move to a dotnet api, I don't think it's worth the effort to change to stime now.  I need this change as the native build portion for Alpine fails here.

File: tools/releaseBuild/Images/microsoft_powershell_alpine3/build-and-run-pwsh.sh [hunk 2]
```diff
@@ -62,13 +72,13 @@ cd ..
...
 cat > $targetFile <<-"EOF"
 <Project>
-  <Target Name="_GetDependencies"
-          DependsOnTargets="ResolveAssemblyReferencesDesignTime">
-    <ItemGroup>
-      <_RefAssemblyPath Include="%(_ReferencesFromRAR.ResolvedPath)%3B" Condition=" '%(_ReferencesFromRAR.Type)' == 'assembly' And '%(_ReferencesFromRAR.PackageName)' != 'Microsoft.Management.Infrastructure' " />
-    </ItemGroup>
-    <WriteLinesToFile File="$(_DependencyFile)" Lines="@(_RefAssemblyPath)" Overwrite="true" />
-  </Target>
+    <Target Name="_GetDependencies"
```
Comment (dantraMSFT) [inline tools/releaseBuild/Images/microsoft_powershell_alpine3/build-and-run-pwsh.sh]: Minor nit: Are the white space diffs really needed?

**Review by dantraMSFT — APPROVED** [review]

File: tools/releaseBuild/Images/microsoft_powershell_alpine3/build-and-run-pwsh.sh (hunk 2 above)
Comment (SteveL-MSFT) [reply tools/releaseBuild/Images/microsoft_powershell_alpine3/build-and-run-pwsh.sh]: Consistent 4 spaces indentation (was copied from build.psm1)

**Review by adityapatwardhan — APPROVED** [review]

**Review by TravisEz13 — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
Filed issue to figure out Alpine Linux release https://github.com/PowerShell/PowerShell/issues/7167

**Comment by SteveL-MSFT** [comment]
I think it's fine to have just an Alpine docker image w/ PSCore6

**Comment by SteveL-MSFT** [comment]
I should change the runtime to `linux-musl-x64` to match the packaging

**Comment by SteveL-MSFT** [comment]
Note that for `dotnet`, alpine-x64 == linux-musl-x64, however the nuget pkg that contains crossgen has it under the `linux-musl-x64` file path, so using that one.

**Pull request merged**